cli/src/anipy_cli/comps/DownloadComp.py

```diff
@@ -0,0 +1,146 @@
+
+from pathlib import Path
+from typing import Any, Callable, List, Tuple
+from anipy_cli.arg_parser import CliArgs
+from anipy_cli.colors import color, colors
+from anipy_cli.config import Config
+from anipy_cli.util import DotSpinner, get_download_path
+
+from anipy_api.anime import Anime
+from anipy_api.download import Downloader
+from anipy_api.provider.base import Episode, LanguageTypeEnum
+
+
+class DownloadComponent:
+    """
+    A component used to download anime for
+    the ani-py CLI.
+    """
+    def __init__(self, cliArgs: CliArgs, dl_path: Path) -> None:
+        self.options = cliArgs
+        self.dl_path = dl_path
+
+    def download_anime(
+        self,
+        picked: List[Tuple[Anime, LanguageTypeEnum, List[Episode]]],
+        /,
+        after_success_ep: Callable[[Anime, Episode, LanguageTypeEnum], Any] = lambda _,__,___: None,
```

Comment:
please use a protocol for the callback as done in for example in the download file in the api